src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,465 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+
+//
+// This captures information of how various key methods of building with MSBuild ran.
+//
+namespace Microsoft.Build.Eventing
+{
+    // /OnlyProviders=*MSBuild-Profiling
+    [EventSource(Name = "Microsoft-Build")]
+    internal sealed class MSBuildEventSource : EventSource
+    {
+        #region Singleton instance
+
+        // define the singleton instance of the event source
```

Comment:
Comment not super useful.

Reply:
I think it's important to note that it's a singleton, i.e., don't make more than you need.